src/Index/MappingBuilder.php

```diff
@@ -25,6 +37,8 @@ public function buildIndexMapping(IndexConfigInterface $indexConfig)
     {
         $mapping = $this->buildMapping($indexConfig->getModel(), $indexConfig);
 
+        $this->dispatcher->dispatch(new PostMappingBuildEvent($indexConfig, $mapping));
+
```

Comment:
```suggestion
$mapping = $event->getMapping()
```

That would allow to tweak the mapping in a listener, requires previous setter to be added

Reply:
@darthf1 something like:

```php
$event = new PostMappingBuildEvent($indexConfig, $mapping);
$this->dispatcher->dispatch($event);

$mapping = $event->getMapping();
```